packages/polling-controller/src/StaticIntervalPollingController.test.ts

```diff
@@ -0,0 +1,242 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import { useFakeTimers } from 'sinon';
+
+import { advanceTime } from '../../../tests/helpers';
+import { StaticIntervalPollingController } from './StaticIntervalPollingController';
+
+const TICK_TIME = 1000;
+
+const createExecutePollMock = () => {
+  const executePollMock = jest.fn().mockImplementation(async () => {
+    return true;
+  });
+  return executePollMock;
+};
+
+class MyGasFeeController extends StaticIntervalPollingController<
+  any,
+  any,
+  any
+> {
+  _executePoll = createExecutePollMock();
+
+  _intervalLength = TICK_TIME;
+}
+
+describe('StaticIntervalPollingController', () => {
+  let clock: sinon.SinonFakeTimers;
+  let mockMessenger: any;
+  let controller: any;
+  beforeEach(() => {
+    mockMessenger = new ControllerMessenger<any, any>();
+    controller = new MyGasFeeController({
+      messenger: mockMessenger,
+      metadata: {},
+      name: 'PollingController',
+      state: { foo: 'bar' },
+    });
+    clock = useFakeTimers();
+  });
+  afterEach(() => {
+    clock.restore();
+  });
+
+  describe('startPollingByNetworkClientId', () => {
+    it('should start polling if not polling', async () => {
+      controller.startPollingByNetworkClientId('mainnet');
+      await advanceTime({ clock, duration: 0 });
+      expect(controller._executePoll).toHaveBeenCalledTimes(1);
+      await advanceTime({ clock, duration: TICK_TIME });
+      expect(controller._executePoll).toHaveBeenCalledTimes(2);
+      controller.stopAllPolling();
+    });
+    it('should call _executePoll immediately and on interval if polling', async () => {
+      controller.startPollingByNetworkClientId('mainnet');
+      await advanceTime({ clock, duration: 0 });
+      expect(controller._executePoll).toHaveBeenCalledTimes(1);
+      await advanceTime({ clock, duration: TICK_TIME * 2 });
+      expect(controller._executePoll).toHaveBeenCalledTimes(3);
+    });
+    it('should call _executePoll immediately once and continue calling _executePoll on interval when start is called again with the same networkClientId', async () => {
```

Comment:
Nit: We know we are testing `startPollingByNetworkClientId` so we don't need to repeat the name of the method in the test:

``` suggestion
    it('should call _executePoll immediately once and continue calling _executePoll on interval when called again with the same networkClientId', async () => {
```

Reply:
done here: https://github.com/MetaMask/core/pull/3636/commits/823aaa188f572a8aa76973a012ae9bfa8da518e0